### DISCUSSION TITLE
Change dropdown direction based on available space

### DISCUSSION THREAD

**Discussion by Laplandia** [add2cal/add-to-calendar-button#92] [Ideas / Requests]
If the button is located at the bottom of the screen, dropdown should probably open up, not down. That way user does not have to scroll the page to see all available options.

**Comment by jekuer** [comment]
Theoretically easy to integrate.
However, this could break the style. In terms of this repo, definitely the 3D-Style + any potential customization on user side.

**Three ideas:**
1. We are planning to include the style a little bit deeper in the js. After that move, we could adjust the position depending on the selected style. This could still break custom adjustments, but acceptable in that case - we might add the option "custom" and then it is up to the developer.
2. Whenever your case is true, we show the modal version instead of the dropdown.
3. Maybe a new config option "dynamicPosition" (default: false). If explicitely set true, we show the dropdown above. In this case, it is up to the developer to activate this behavior. It would not break things for others, but who wants to optimize can do so. We could also add specific additional classes for that case, which can be used in the CSS to adjust styling.

I would prefer 1+3. Waiting for the refactoring to better distinguish between styles. Then add the new option, where we set the default based on the styling.

Ok - this was more of a rubber duck session 😆 . Putting it on the roadmap, but a little further down this road.

**Reply by Laplandia** [reply]
Thanks! 

I personally much prefer 1+3. Modal interaction does not fit with the rest of our UX.

Another option that is completely different is to expose a method that just produces links and files. Then it could be used with any custom UI.

**Reply by jekuer** [reply]
Good news!
Will be supported with latest PR #114 and upcoming version 1.14

**Reply by Laplandia** [reply]
Does this logic only apply when the default button is used? Can it by applied for custom buttons?

**Reply by jekuer** [reply]
It is universal, but requires the generated button. If you create your own button and use the atcb_action method, it would not work. Basically, because it is hard to control in a stable way, when the style of the button is not controlled by the script.

We will add more options to control the style of the generated button - but this takes some more time.

**Reply by Laplandia** [reply]
Thanks a lot! Looking forward to it.
